### PR TITLE
docs: Consistent type names in RPC help descriptions

### PR DESCRIPTION
**Pull request by ch4ot1c**
Followup to https://github.com/bitcoin/bitcoin/pull/14373.

Now, only these appear in description text:
```
boolean
numeric
string

json array
json array of xs
json object
json object of xs
```

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -684,12 +684,12 @@ static UniValue getbalance(const JSONRPCRequest& request)
 
     if (request.fHelp || (request.params.size() > 3 ))
         throw std::runtime_error(
-            "getbalance ( \"(dummy)\" minconf include_watchonly )\n"
+            "getbalance ( dummy minconf include_watchonly )\n"
```
Comment (promag) [inline src/wallet/rpcwallet.cpp]: `dummy` is a string.

**Comment by ch4ot1c** [comment]
#### Final notes:

(numeric) or (int)
(boolean)
(string)

(x, optional)
(x, optional, default=y)
(x, optional, default=\"y\")

(x, required [for ...])

(array)
(array of x[s])
(object)

Only these appear as types now, in both in requests and results.

Additional notes:
- Any usage of an ` = ` within RPC descriptions has been trimmed
- Suffixed `Default: ` cases have been moved to the type string e.g. `(x, optional, default=y)` in RPCs
- The json object for label & purpose data is now named 'Address information' in both `getaddressinfo` and `getaddressesbylabel`

- `Must be one of:\n prefix should be present for all argument option lists >= 3
- `Results` still has different/inconsistent formatting throughout, which will need to be addressed separately.

**Comment by MarcoFalke** [comment]
* importprunedfunds has arguments, but they are not listed in the help string. Mind adding them?
* uptime takes (and ignores) a dummy argument, whereas it shouldn't. Mind fixing the code to throw when a dummy is provided?

File: src/rpc/server.cpp [hunk 2]
```diff
@@ -245,6 +245,11 @@ static UniValue uptime(const JSONRPCRequest& jsonRequest)
                 + HelpExampleRpc("uptime", "")
         );
 
+    const UniValue& dummy_value = jsonRequest.params[0];
+    if (!dummy_value.isNull()) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "No dummy first argument may be included.");
+    }
+
```
Comment (MarcoFalke) [inline src/rpc/server.cpp]: Can be fixed instead by replacing `> 1` with `> 0` a few lines up?

**Comment by MarcoFalke** [comment]
`prioritisetransaction` has an argument with a space. Mind to replace the space with underscore?

File: src/wallet/rpcdump.cpp [hunk 3]
```diff
@@ -1108,8 +1113,8 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
             "importmulti \"requests\" ( \"options\" )\n\n"
```
Comment (MarcoFalke) [inline src/wallet/rpcdump.cpp]:
```suggestion
             "importmulti \"requests\" ( \"options\" )\n"
```

File: src/rpc/server.cpp (hunk 2 above)
Comment (ch4ot1c) [reply src/rpc/server.cpp]: Ah, that was what I thought at first.

**Comment by ch4ot1c** [comment]
`prioritisetransaction` and `getbalance` changes are already in the PR.

File: src/zmq/zmqrpc.cpp [hunk 4]
```diff
@@ -20,7 +20,7 @@ UniValue getzmqnotifications(const JSONRPCRequest& request)
             "\nReturns information about the active ZeroMQ notifications.\n"
             "\nResult:\n"
             "[\n"
-            "  {                        (json object)\n"
+            "  {                        (object)\n"
```
Comment (MarcoFalke) [inline src/zmq/zmqrpc.cpp]: Half of the changes seem to be of this fashion. Mind to create a primitive scripted-diff commit in a separate pull request to get the easy to review things out of the way?

Comment (ch4ot1c) [reply src/zmq/zmqrpc.cpp]: Damn, they need to make a `sed`-builder plugin alongside find-and-replace in vscode, that supports negation. Way easier workflow for newer contributors than all the nuanced escape characters that are necessary in *nix tools.

Is it preferred to use `git grep (-l) (-L)` or `find . x` in scripted diffs?

Also, does this really make it easier to review for correctness? I'm not so sure.

Next commit will be a scripted diff for:
> Replace `(boolean)W` with `(boolean) W` in `.`

`git grep -l "(boolean)W" | xargs sed -i "s/(boolean)W/(boolean) W/g"`
is about right I think?

What if I'm stuck in PowerShell! No `xargs` available.

Comment (ch4ot1c) [reply src/zmq/zmqrpc.cpp]: And on Mac, `gsed` is needed to replicate `sed` behavior performed in scripted-diffs

Comment (MarcoFalke) [reply src/zmq/zmqrpc.cpp]: I'd do it like this: `sed -i --regexp-extended -e 's/\(json (array|object)\)/(\1)/g' $(git grep -l '(json ')`. There might be a hack to make it work on macOS, but we mostly care that it runs on Ubuntu et al.

**Comment by ch4ot1c** [comment]
Pushed to https://github.com/ch4ot1c/bitcoin/tree/fix/rpc-descriptions, but the commit isn't appearing here. Strange...

File: src/zmq/zmqrpc.cpp (hunk 4 above)
Comment (ch4ot1c) [reply src/zmq/zmqrpc.cpp]: Pushed to https://github.com/ch4ot1c/bitcoin/tree/fix/rpc-descriptions, but the commit isn't appearing here. Strange...

**Comment by MarcoFalke** [comment]
Could rebase and squash with something like:

```sh
git fetch bitcoin
git checkout fix/rpc-descriptions
git merge bitcoin/master
git reset --soft bitcoin/master
git commit -m "More RPC help description fixes"
git push origin fix/rpc-descriptions --force

**Comment by laanwj** [comment]
utACK 2f6fff743a2f1e7c51bee41c18498b4043ba9e48

**Comment by MarcoFalke** [comment]
`src/rpc/util.cpp` uses "json object". Imo everything should use the same

**Comment by ch4ot1c** [comment]
@MarcoFalke I see. To confirm, `json array` and `json object` everywhere applicable, preferred over `array` and `object`, aka `util.cpp` is set in stone? Will redo as a scripted-diff once more.

File: src/rpc/mining.cpp [hunk 5]
```diff
@@ -314,12 +314,12 @@ static UniValue getblocktemplate(const JSONRPCRequest& request)
                     {"template_request", RPCArg::Type::OBJ, /* opt */ true, /* default_val */ "", "A json object in the following spec",
                         {
                             {"mode", RPCArg::Type::STR, /* opt */ true, /* default_val */ "", "This must be set to \"template\", \"proposal\" (see BIP 23), or omitted"},
-                            {"capabilities", RPCArg::Type::ARR, /* opt */ true, /* default_val */ "", "A list of strings",
+                            {"capabilities", RPCArg::Type::ARR, /* opt */ true, /* default_val */ "", "An array of strings",
```
Comment (ch4ot1c) [inline src/rpc/mining.cpp]: `A json array` here instead?

**Comment by laanwj** [comment]
Concept ACK.

Though I think the jargon switch to `numeric` loses information.

The code, in practice, makes a difference between `numeric` (in general) and `integer`. For example where `get_int` is used, only an integer will be accepted. Not a decimal value with a point.

**Pull request closed without merging**